File: LTNC/STL_Stack_Queue/DOICS10_2.cpp
#include <bits/stdc++.h>
using namespace std;

struct Stack {
    vector<char> a;
    void push(char x){ a.push_back(x); }
    char pop(){ char x=a.back(); a.pop_back(); return x; }
    bool empty() const { return a.empty(); }
    char top() const { return a.back(); }
};

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    long long n;
    if(!(cin>>n)) return 0;
    if(n==0){
        cout<<0;
        return 0;
    }
    Stack st;
    while(n>0){
        st.push(char('0'+(n%2)));
        n/=2;
    }
    string res;
    while(!st.empty()) res.push_back(st.pop());
    cout<<res;
    return 0;
}

File: LTNC/STL_Stack_Queue/DOICS10_2.py
'''
Yêu cầu xây dựng 1 lớp Ngăn xếp, sau đó áp dụng vào giải bài tập sau:
Nhập vào 1 số nguyên ở hệ 10, yêu cầu xuất ra 1 số ở hệ 2

Ví dụ:

Input
10
Output
1010

'''

import sys

class Stack:
    def __init__(self):
        self.a=[]
    def push(self,x):
        self.a.append(x)
    def pop(self):
        return self.a.pop()
    def empty(self):
        return not self.a
    def top(self):
        return self.a[-1]

data=sys.stdin.read().strip().split()
if not data: sys.exit()
n=int(data[0])
if n==0:
    print(0)
    sys.exit()
st=Stack()
while n>0:
    st.push(str(n%2))
    n//=2
res=[]
while not st.empty():
    res.append(st.pop())
print("".join(res))